src/print.c

```diff
@@ -245,6 +255,190 @@ launch_preview (const char *filename,
   return retval;
 }
 
+static int
+pdf_get_actions_page (char        *filename,
+                      PDF_ACTIONS  act,
+                      int          page)
+{
+  GSubprocess *process;
+  GInputStream *stream;
+  char *option1 = NULL;
+  char *option2 = NULL;
+  char buffer[50] = { 0 };
+
+  option1 = g_strconcat ("--file=", filename, NULL);
+  switch(act)
+  {
+    case PDF_TEST :
+       (void)page;
+       option2 = g_strdup ("--test");
+       break;
+    case PDF_NUM_PAGES :
+       (void)page;
+       option2 = g_strdup ("--pages");
+       break;
+    case PDF_WIDTH :
+       option2 = g_strdup_printf ("--width=%d", page);
+       break;
+    case PDF_HEIGHT :
+       option2 = g_strdup_printf ("--height=%d", page);
```

Comment:
These are leaked. I also think you should be using `g_subprocess_newv()` to pass an argv.

Reply:
Fixed